### PR TITLE
feat: add Lint rule to detect wrong string annotation usages

### LINKED ISSUE #917: [FR]: xml parsing in a lint rule for our custom annotations
### 🕵️ Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the problem

User would miss spell our custom annotation names like `typography` `large`

### Describe the solution

building a lint rule that will assert that only supported values are used for our keys 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by EliottLujan**
## 📋 Changes

- Add a custom lint rule to detect wrong attribute names and values when using `annotations` for string
- Detect wrong `variable` usages
- Add unit tests for the new lint rule

## 🤔 Context

Closes #917 

~Note that this Lint rule doesn't detect when using empty annotations as Simon mentioned [here](https://github.com/adevinta/spark-android/issues/917#issuecomment-1908469355). If you guys think this PR can be merged, I'll create a different Github issue to ensure we also plan to have a detector for empty annotations~

## ✅ Checklist

- [x] I have reviewed the submitted code.

### CODE REVIEW

File: spark-lint/src/main/kotlin/com/adevinta/spark/lint/LintIssueRegistry.kt [hunk 1]
```diff
@@ -36,5 +36,7 @@ public class LintIssueRegistry : IssueRegistry() {
     override val api: Int = CURRENT_API
     override val issues: List<Issue> = listOf(
         MaterialComposableUsageDetector.ISSUE,
+        UnknownStringResourceAnnotationsDetector.UNKNOWN_ANNOTATION_NAME_ISSUE,
+        UnknownStringResourceAnnotationsDetector.UNKNOWN_ANNOTATION_VALUE_ISSUE,
```
Comment (EliottLujan) [inline spark-lint/src/main/kotlin/com/adevinta/spark/lint/LintIssueRegistry.kt:40]: I wasn't sure whether to go for
- 1 detector and 2 issues
- 2 detectors and 1 issue each
- 1 detector and 1 issue

I felt like having 2 issues for a single detector is fine, let me know what you think!

File: spark-lint/src/main/kotlin/com/adevinta/spark/lint/UnknownStringResourceAnnotationsDetector.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        )
+
+        val UNKNOWN_ANNOTATION_NAME_ISSUE = Issue.create(
+            id = "UnknownStringResourceAnnotationsDetector",
+            briefDescription = "Unknown annotation name",
+            explanation = "This annotation name is not supported and won't be parsed. " +
+                "Supported names are $VALID_ANNOTATION_NAMES",
+            category = CORRECTNESS,
+            priority = 8,
+            severity = ERROR,
```
Comment (EliottLujan) [inline spark-lint/src/main/kotlin/com/adevinta/spark/lint/UnknownStringResourceAnnotationsDetector.kt:126]: Feel free to let me know if you think the priority or severity should be changed!

File: spark-lint/src/test/kotlin/com/adevinta/spark/lint/UnknownStringResourceAnnotationsDetectorTest.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import com.adevinta.spark.lint.UnknownStringResourceAnnotationsDetector.Companion.UNKNOWN_ANNOTATION_VALUE_ISSUE
+import com.android.tools.lint.checks.infrastructure.LintDetectorTest
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+public class UnknownStringResourceAnnotationsDetectorTest : LintDetectorTest() {
```
Comment (EliottLujan) [inline spark-lint/src/test/kotlin/com/adevinta/spark/lint/UnknownStringResourceAnnotationsDetectorTest.kt:34]: The existing `MaterialComposableUsageDetectorTest` has the `public` visibility modifier on its class and methods, so I also added it for `UnknownStringResourceAnnotationsDetectorTest` but I'm not sure we actually need it?

**Review by SimonMarquis — COMMENTED** [review]
Thanks for this PR 🙏
I'll push the many changes I've commented on, on top of your review.

File: spark-lint/src/test/kotlin/com/adevinta/spark/lint/UnknownStringResourceAnnotationsDetectorTest.kt (hunk 3 above)
Comment (SimonMarquis) [reply spark-lint/src/test/kotlin/com/adevinta/spark/lint/UnknownStringResourceAnnotationsDetectorTest.kt]: It seems like our `explicitApi` is not configured (maybe on purpose?) on test sources.

**Review by soulcramer — APPROVED** [review]
Thanks for the contribution it helps a lot 😊!

**Pull request merged**